rfd/00xx-pod-access-requests.md

```diff
@@ -0,0 +1,555 @@
+dep-2 pod     node:dep-2
+
+$ tsh request create --resources "pod:pod1,pod:dep-*"
+Waiting for request to be approved...
+Approved!
+$ kubectl get pods 
+```
+
+Similarly to Resource Based Access Requests, Pod Access requests will leverage
+`search_as_roles` to control which Pods the user can view while searching.
+
+```yaml
+kind: role
+metadata:
+  name: response-team
+spec:
+  allow:
+    request:
+      # search_as_roles allows a member of the response team
+      # to search for resources accessible to users with the kube-admins role,
+      # which they will be allowed to request access to. This includes Pods in 
+      # the target Kubernetes Cluster.
+      search_as_roles: [kube-admin]
+```
+
+```yaml
+kind: role
+metadata:
+  name: kube-admin
+spec:
+  allow:
+    # kubernetes_groups will be used by Teleport to fetch the Pods available in 
+    # the target cluster. The kubernetes_groups field is used by Kubernetes RBAC 
+    # to limit the Pods the user can request access to.
+    kubernetes_groups: ["system:masters"]
+    # kubernetes_labels defines which Kubernetes clusters the user is able to 
+    # request Pod access into.
+    kubernetes_labels:
+       owner: prod_team
+```
+
+While requesting access to a Pod or searching for Pods, the user must identify
+which cluster he wants to request access to. Request access to multi-clusters
+will not be supported.
+
+For now, all Pod access requests come from `tsh`, but the functionality can be
+extended in the future using a `kubectl` plugin.
+
+## Option 1: Restrict Pod access
+
+Proposed is a new way to restrict access to specific Pods on a Kubernetes cluster.
+Given a set of roles with Kubernetes RBAC principals - groups or users,
+that normally grant access to a larger set of Pods, once `kubernetes_pods`
+is specified in at least one of the roles, Teleport will restrict the user's
+access to the list of pods specified under `kubernetes_pods`.
+`kubernetes_pods` must define a subset of the Pods allowed by the user's Kubernetes
+RBAC principals.
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: my-kube-role
+spec:
+  allow:
+    kubernetes_labels:
+      "*": "*"
+    kubernetes_groups: ["kube_group"]
+    kubernetes_pods: ["B","C", "podname-*-*"]
+```
+
+Having a Kubernetes RBAC group `kube_group` allowing access to a set of Pods
+{A,B,C,D, podname-1-1} and a Teleport role with `kubernetes_pods` set to
+{B,C,podname-\*-\*}, a request would have the following behavior:
+
+- `kubectl get pods`: returns pods {B,C,podname-1-1}.
+- `kubectl edit pod/B`: request allowed.
+- `kubectl edit pod/A`: request denied by Teleport.
+- `kubectl delete pod/B`: request allowed.
+- `kubectl logs pod/B`: request allowed.
+- `kubectl logs pod/A`: request denied by Teleport.
+- `kubectl logs pod/podname-1-1`: request allowed.
+
+Using this method, Teleport allows you to restrict the pods that a particular
+user has access to. However, if there is at least one role that specifies a pod
```

Comment:
Same as above: clarify this applies to roles matching the cluster you're accessing. Otherwise it reads like it will have a poison pill behavior.